src/wms/src/main/java/org/geoserver/wms/kvp/FilterKvpParser.java

```diff
@@ -73,46 +74,70 @@ public Object parse(String value) throws Exception {
                 filters.add(Filter.INCLUDE);
             } else {
                 Filter filter;
+                String exceptionMsg = "";
                 final byte[] rawContent = string.getBytes();
                 InputStream input = new ByteArrayInputStream(rawContent);
-
                 try {
-                    // create the parser
-                    Configuration configuration = new OGCConfiguration();
-                    Parser parser_1_0_0 = new Parser(configuration);
-                    parser_1_0_0.setEntityResolver(resolverProvider.getEntityResolver());
-                    filter = (Filter) parser_1_0_0.parse(input);
+                    if (string.toLowerCase().contains(FES.NAMESPACE)) {
+                        filters.add(parseOGCFilter_2_0_0(input));
```

Comment:
Looking at how the code flows, it would seem that if a filter happens to be parseable as both V2 and V1.1 or V1.0, or the legacy parser, then it would be added to filters two times? I don't see a test that the filter has been parsed once already... if V2 parsing works, then the other try should not be run. I'm guessing that the other attempts shoud be done only if pasing V2 fails with an excepion (that is, in its catch branch).